### PR TITLE
Explore: List query templates

### LINKED ISSUE #86122: Explore: Load Query Library items

### PR DESCRIPTION
**Pull request by ifrost**
Fixes #86122

For testing please use [Query Library App plugin](https://github.com/grafana/grafana-querylibrary-app/issues/3), or create some templates by [posting requests directly to the API](https://github.com/grafana/grafana/issues/81357):

https://github.com/grafana/grafana/assets/745532/8187e227-f5f2-4739-b3fd-9d9a8d259670

Changes:
- Shows the list of query templates
- Displayed columns:
  - the name of the data source, title, and the query
  - data source type
  - date added
- Basic error handling

Out of scope:
- Displaying the user who created the query template (#86987)
- Query template actions (#83097, #84895, #84896, #84900, #84897, #86986)

Follow-ups:
- #86960
- #86838

### CODE REVIEW

**Review by gelicia — APPROVED** [review]
Scaffolding looks great, confirmed by adding a query to the library app and seeing it in Explore! Exciting stuff!

**Comment by ifrost** [comment]
@grafana/grafana-frontend-platform I'm not 100% how to correctly add a dependency to `grafana/runtime`. What I'm trying to achieve is to expose the `ApiProvider` and React hooks from RTK Query to plugins. I want to use it in core (Explore) and in an Application Plugin (Query Library). It requires `reactjs/toolkit` (for RTK Query) to work, which depends on on a peer dependency of `react-redux`, which depends on `react-is`. If I don't add @reactjs/redux or react-redux to dependencies/peerDependencies in `grafana-runtime `I get this error when bulding `grafana-runtime`:

```
[!] Error: 'isValidElementType' is not exported by ../../node_modules/react-is/index.js, imported by ../../node_modules/react-redux/es/components/connect.js
https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module
../../node_modules/react-redux/es/components/connect.js (8:9)
 6: import hoistStatics from 'hoist-non-react-statics';
 7: import * as React from 'react';
 8: import { isValidElementType, isContextConsumer } from 'react-is';
             ^
 9: import defaultSelectorFactory from '../connect/selectorFactory';
10: import { mapDispatchToPropsFactory } from '../connect/mapDispatchToProps';
Error: 'isValidElementType' is not exported by ../../node_modules/react-is/index.js, imported by ../../node_modules/react-redux/es/components/connect.js
    at error (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:198:30)
    at Module.error (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:12560:16)
    at Module.traceVariable (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:12919:29)
    at ModuleScope.findVariable (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:11571:39)
    at FunctionScope.findVariable (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:6503:38)
    at ChildScope.findVariable (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:6503:38)
    at ReturnValueScope.findVariable (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:6503:38)
    at ChildScope.findVariable (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:6503:38)
    at Identifier.bind (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:7570:40)
    at CallExpression.bind (/Users/ifrost/projects/grafana/node_modules/rollup/dist/shared/rollup.js:5400:23)
```

The error is coming from rollup which should be able to resolve the dependency from the parent but it fails 🤔 


Not sure why though, maybe because react-is resolves dependency conditionally? 

```
if (process.env.NODE_ENV === 'production') {
  module.exports = require('./cjs/react-is.production.min.js');
} else {
  module.exports = require('./cjs/react-is.development.js');
}
```

Adding dependency/peerDependency cause the module to be marked as external (via externals rollup plugin) and it's bundled correctly, but I'm not sure if that's the right way of doing this 🤔

**Comment by gelicia** [comment]
Had a chat with @ashharrison90 and @joshhunt about this - unfortunately this PR is in the middle of two unknown and emerging practices that the frontend platform team is working on - both app platform and a common interface for generating endpoints. They would ideally like some of the code that is handwritten in here to be generated, to provide guardrails and shape to minimize the possibility of API instability between releases. 

If this ends up taking long enough where it becomes a blocker, we can look into marking the query library API as internal and making extremely clear it is unstable and moving forward with it. I communicated to them that I don't consider myself blocked at the moment and it could wait, and I would just work off this branch. 

Cc: @ifrost

**Comment by ifrost** [comment]
After having a chat with @grafana/grafana-frontend-platform we have decided to move this functionality back to core and used it from there until there's a consenus reached on how such APIs should be exposed.

Related issues/prs/docs:
* #84494
* #87540
* #87175 
* https://docs.google.com/document/d/1Ite1CVu2xqOVGSKIgR-sQWdbJpwFE_pLMuXPxEufL9c/edit#heading=h.5sybau7waq2q

File: public/app/features/query-library/api/query.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { BaseQueryFn } from '@reduxjs/toolkit/query/react';
+import { lastValueFrom } from 'rxjs';
+
+import { getBackendSrv, isFetchError } from '@grafana/runtime/src/services/backendSrv';
+
+import { DataQuerySpecResponse } from './types';
+
+export const BASE_URL = '/apis/peakq.grafana.app/v0alpha1/namespaces/default/querytemplates/';
+
+/**
```
Comment (gelicia) [inline public/app/features/query-library/api/query.ts:10]: I might put some warnings in here about how it is alpha and an unstable API.

**Review by gelicia — APPROVED** [review]

File: public/app/features/query-library/api/query.ts (hunk 1 above)
Comment (ifrost) [reply public/app/features/query-library/api/query.ts:15]: Added in [bc7f1c7](https://github.com/grafana/grafana/pull/86897/commits/bc7f1c7f99b5072ed1aff73c762320a5f19fd2e5)

**Pull request merged**